### ISSUE TITLE
Custom sort function

### ISSUE THREAD

**Issue by iafisher** [grid-js/gridjs#48]
Cool project! It would be really useful if you could specify a custom sort function, similar to the `compareFunction` parameter of [`Array.sort`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort).

```javascript
function compareNumbers(a, b) {
    return a - b;
}

const grid = new Grid({
  columns: [
    {name: 'Name'},
    {name: 'Email'},
    {name: 'Salary', sort: true, compareFunction: compareNumbers},
  ],
  data: [
    ['John', 'john@example.com', '4000'],
    ['Mark', 'mark@gmail.com',  '5000'],
    ['Jane', 'jane@gmail.com',   '45000'],
  ]
});
```

Otherwise things like numbers and dates won't sort properly.

**Comment by balexandre** [comment]
but if you send a numeric value, it will order numbers correctly, like

```javascript
[
    ['John', 'john@example.com', 4000],
    ['Mark', 'mark@gmail.com',  5000],
    ['Jane', 'jane@gmail.com',   45000]
]
```

**Comment by iafisher** [comment]
Fair enough, but it still doesn't work for most date formats, or for columns of numbers loaded from an existing HTML table.

**Comment by afshinm** [comment]
makes sense! I will release a version today which will add custom comparing function to columns.

**Comment by afshinm** [comment]
@iafisher @balexandre fixed in 1.3.0, see https://gridjs.io/docs/examples/custom-sort

**Comment by iafisher** [comment]
Great, thank you!
